force-app/main/classes/FFClient.cls

```diff
@@ -58,7 +58,7 @@ public class FFClient {
     private List<FFModelsFeatures.TargetSegment> getTargetSegments() {
         FFCache cache = this.config.cache;
         List<FFModelsFeatures.TargetSegment> targetSegments = (List<FFModelsFeatures.TargetSegment>) cache.getSegments();
-        if(targetSegments == null) {
```

Comment:
This works for us since we have target groups in our setup. 
But if some one runs this in an environment with no segments this would end up calling the API on every variation call.

Reply:
@andybharness Pl note the PR is not complete as the fix only works if the environment has atleast one segment. Also unit-tests are not updated.